Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/Interactable.cs

```diff
@@ -127,25 +140,38 @@ protected static IMixedRealityInputSystem InputSystem
 
         // IInteractableEvents
         protected List<IInteractableHandler> handlers = new List<IInteractableHandler>();
+
         protected Coroutine globalTimer;
-        protected float clickTime = 0.3f;
-        protected Coroutine inputTimer;
 
-        // reference to the pointer action that started an interaction
-        protected MixedRealityInputAction pointerInputAction;
+        // 
+        // Clicking
+        //
 
+        // A click must occur within this time after an input down
+        protected float clickTime = 0.5f;
+        protected Coroutine clickValidTimer;
         // how many clicks does it take?
         protected int clickCount = 0;
-
-        // cache a drag value
-        protected Vector3 dragStart = Vector3.zero;
-
         /// <summary>
         /// how many times this interactable was clicked
         /// good for checking when a click event occurs.
         /// </summary>
         public int ClickCount => clickCount;
 
+        // 
+        // Variables for determining gesture state
+        //
+        protected Vector3? dragStart3 = null;
+        protected Vector2? dragStart2 = null;
```

Comment:
A bit more documentation here and maybe reflect in the naming that they are for 2D and 3D drag.

Reply:
Done! I also refactored / cleaned up that code to remove duplication.